### ISSUE TITLE
List with connection resetted

### ISSUE THREAD

**Issue by feldti** [jgfoster/Jade#246]
Another pretty annoying problem is, that the list of Gemstone-Server (Jade Login) is very often reseted, so that only the default entry is available.

**Comment by feldti** [comment]
This is for example done, if I finish a Jade Login window and a Jade is running

**Comment by thelliez** [comment]
Yes same problem here.  In case of Jade crash (and it happens fairly often on Windows 10 running on top of VMWare Fusion on a Mac), all the Jade login info are lost.  I have to rekey all the session info.

**Comment by jgfoster** [comment]
The user configuration data is stored by JadeLoginShell>>saveData at %USERPROFILE%\AppData\Roaming\JadeData.gss (based on the registry entry at Computer\HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Explorer\User Shell Folders\AppData). It is read when JadeLoginShell opens and if there is an error opening or reading the file then we use defaults. 

I've tried killing Jade (using the Task Manager) and it doesn't harm the file. Any ideas of how it could become corrupt? Perhaps Jade dies when trying to write it?

As a proposed fix I'll write a new file, delete the old one, and then rename the new one to the expected name.

**Comment by jgfoster** [comment]
The code in 2.3.16 tries to handle crash during writing the data.

**Comment by jgfoster** [comment]
@feldti in #278: 

> Perhaps it would be nice, if the connection information are not saved automatically (somehow), but by using a simple button. I am still loosing (again and again) all my connection informations over the time and have to enter them again and again.
>
>Today at least three times this morning - a simple button would be nice and no automatically saving

**Comment by jgfoster** [comment]
@feldti, To understand this better, could you try launching Jade and, if the connection list is good, look at the path shown in my "May 3, 2020" comment and make a copy of that file. Then, when the connection list is bad quit Jade, copy the file back into the provided location and relaunch Jade. It would also help to see what the last modified timestamp is on the file and see if we can correlate it with a launch or quit of Jade. Thanks!
